src/app/shared/utilities/util.ts

```diff
@@ -70,9 +74,20 @@ export class Utility {
     return Promise.all([sourceId, projectId, pariticipantId])
   }
 
-  getLatestKafkaSchemaVersions(specs) {
-    const qKey = specs.avsc + '_' + specs.name + '-key'
-    const qVal = specs.avsc + '_' + specs.name + '-value'
+  getKafkaTopic(type) {
+    return this.http
+      .get(DefaultSchemaSpecEndpoint, { responseType: 'text' })
+      .toPromise()
+      .then(res => {
+        const schemaSpecs = YAML.parse(res).data
+        type = type.toLowerCase()
+        return schemaSpecs.find(t => t.type.toLowerCase() == type).topic
+      })
```

Comment:
Also add a fallback in case the `type` is not present in the specifications. In that case, it should revert to using the topic name constructed from `avsc` and `name` in the protocol

Reply:
Yup made the changes.